packages/flutter_tools/lib/src/device.dart

```diff
@@ -74,23 +73,15 @@ class DeviceManager {
     IOSSimulators(),
     FuchsiaDevices(),
     FlutterTesterDevices(),
-  ] + _conditionalDesktopDevices + _conditionalWebDevices);
-
-  /// Only add desktop devices if the flag is enabled.
-  static List<DeviceDiscovery> get _conditionalDesktopDevices {
-    return flutterDesktopEnabled ? <DeviceDiscovery>[
+    if (featureFlags.isMacOSEnabled)
```

Comment:
Do we actually need any of these checks? The devices object will say it can't discover devices anyways, right? Having the checks consolidated in fewer places seems easier.

Reply:
Done